src/sseq.rs

```diff
@@ -160,6 +160,14 @@ impl<const N: usize> Iterator for SSeqOneHammingIter<N> {
             Some(next_sseq)
         }
     }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        if self.position >= self.source.len() {
+            (0, Some(0))
+        } else {
+            (0, Some(self.source.len() - self.position))
+        }
+    }
```

Comment:
Hmm, the upper bound is not correct. If the sequence is of length n `SSeqOneHammingIter` could produce upto `4 *n` elements. The exact number depends on the configuration. I don't see a specific reason to implement this function.